tests/rptest/tests/follower_fetching_test.py

```diff
@@ -164,3 +169,79 @@ def test_basic_follower_fetching(self, read_from_object_store):
                     assert current_bytes_fetched > 0
                 else:
                     assert current_bytes_fetched == 0
+
+
+class IncrementalFollowerFetchingTest(PreallocNodesTest):
+    def __init__(self, test_context):
+        super(IncrementalFollowerFetchingTest,
+              self).__init__(test_context=test_context,
+                             num_brokers=3,
+                             node_prealloc_count=1,
+                             extra_rp_conf={
+                                 'enable_rack_awareness': True,
+                             })
+
+    @skip_debug_mode
+    @cluster(num_nodes=5)
+    def test_incremental_fetch_from_follower(self):
+        rack_layout_str = "ABC"
+        rack_layout = [str(i) for i in rack_layout_str]
+
+        for ix, node in enumerate(self.redpanda.nodes):
+            extra_node_conf = {
+                'rack': rack_layout[ix],
+                'enable_rack_awareness': True,
+            }
+            self.redpanda.set_extra_node_conf(node, extra_node_conf)
+
+        self.redpanda.start()
+        topic = TopicSpec(partition_count=12, replication_factor=3)
+
+        self.client().create_topic(topic)
+        msg_size = 512
+
+        producer = KgoVerifierProducer(self.test_context,
+                                       self.redpanda,
+                                       topic,
+                                       msg_size,
+                                       1000000,
+                                       self.preallocated_nodes,
+                                       rate_limit_bps=256 * 1024)
+
+        producer.start()
+        consumer_group = "kafka-cli-group"
+
+        # We are using kafka cli consumer to control metadata, age and client rack id consumer properties
+        cli_consumer = KafkaCliConsumer(self.test_context,
+                                        self.redpanda,
+                                        topic.name,
+                                        group="kafka-cli-group",
+                                        consumer_properties={
+                                            "client.rack": "A",
+                                            "metadata.max.age.ms": 10000
+                                        })
+        cli_consumer.start()
+        cli_consumer.wait_for_messages(100)
+        # sleep long enough to cause metadata refresh on the consumer
+        time.sleep(30)
```

Comment:
q: I'm curious how this test is forcing the client to do  'incremental fetches' vs the other test that  looks pretty similar.

Reply:
This test is using Java Kafka client that by default uses incremetal fetch requests.